src/transformers/models/conditional_detr/modeling_conditional_detr.py

```diff
@@ -0,0 +1,2593 @@
+# coding=utf-8
+# Copyright 2022 Attn4Vis and The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch Conditional DETR model."""
+
+
+import math
+import random
+from dataclasses import dataclass
+from typing import Dict, List, Optional, Tuple
+
+import torch
+from torch import Tensor, nn
+from torch.nn import functional as F
+
+from ...activations import ACT2FN
+from ...modeling_outputs import BaseModelOutput, BaseModelOutputWithCrossAttentions, Seq2SeqModelOutput
+from ...modeling_utils import PreTrainedModel
+from ...pytorch_utils import torch_int_div
+from ...utils import (
+    ModelOutput,
+    add_start_docstrings,
+    add_start_docstrings_to_model_forward,
+    is_scipy_available,
+    is_timm_available,
+    is_vision_available,
+    logging,
+    replace_return_docstrings,
+    requires_backends,
+)
+from .configuration_conditional_detr import ConditionalDetrConfig
+
+
+if is_scipy_available():
+    from scipy.optimize import linear_sum_assignment
+
+if is_vision_available():
+    from .feature_extraction_conditional_detr import center_to_corners_format
+
+if is_timm_available():
+    from timm import create_model
+
+logger = logging.get_logger(__name__)
+
+_CONFIG_FOR_DOC = "ConditionalDetrConfig"
+_CHECKPOINT_FOR_DOC = "Atten4Vis/ConditionalDETR"
+
+CONDITIONAL_DETR_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "Atten4Vis/ConditionalDETR",
+    # See all conditional_detr models at https://huggingface.co/models?filter=conditional_detr
+]
+
+
+@dataclass
+class ConditionalDetrDecoderOutput(BaseModelOutputWithCrossAttentions):
```

Comment:
```suggestion
# Copied from transformers.models.detr.modeling_detr.DetrDecoderOutput with DETR -> Conditional DETR
class ConditionalDetrDecoderOutput(BaseModelOutputWithCrossAttentions):
```

Reply:
This function is actually different from DETR's. Conditional DETR has an additional output 'reference_points'. So we should't add copied from here.

Reply:
Maybe you can copy from Deformable DETR now?

Reply:
Hmmm, I took a look at the Deformable DETR's code. It is still different. Deformable DETR has more output elements than Conditional DETR. I do not think it is a good idea to copy from Deformable DETR either.

Reply:
Ok, makes sense, will close this!